build.assets/tooling/cmd/build-apt-repos/apt_repo_tool.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"io/fs"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type AptRepoTool struct {
+	config       *Config
+	aptly        *Aptly
+	s3Manager    *S3manager
+	supportedOSs map[string][]string
+}
+
+// Instantiates a new apt repo tool instance and performs any required setup/config.
+func NewAptRepoTool(config *Config, supportedOSs map[string][]string) (*AptRepoTool, error) {
```

Comment:
Why not make `supportedOSs` a part of the `Config`?

Reply:
When writing this I thought that would be a great idea, but ultimately decided against it as it has several downsides:
* The `flags` library does not support it, so implementing a `map[string][]string` input would likely add several hundred lines of parsing and code
* Whatever format I went with would make reading the `.drone.yml` and related dronegen function more difficult, and would make adding/removing OS and OS versions more difficult, or would make the dronegen function more brittle

Not pulling in the `supportedOSs` map makes the code overall easier to read and maintain, and pushes the burden of validating the map to the compiler.